MLS.tex

```diff
@@ -115,9 +115,6 @@
 % https://tex.stackexchange.com/questions/370384/change-toc-depth-mid-document
 \addtocontents{toc}{\setcounter{tocdepth}{0}}
 
-% Glossary
-\include{chapters/glossary}
```

Comment:
I'm not sure that we should just remove the glossary.
Obviously having both an index and a glossary would be odd, but as I understand it a latex-glossary normally contain back-links to the use of the term; so one possibility would be to replace the glossary-chapter by a set of glossary-entries and instead of building a document index we would technically build a glossary with index-entries.
(Assuming this isn't what is done already.)

Reply:
As you say, having both an index and a glossary would be odd; it would make it hard to read the document efficiently if some terms are defined in a glossary, while other terms have to be searched in the index.  Therefore, I have tried to carefully migrate the glossary entries into the index, sometimes moving useful phrases from the glossary into the sections where terms are introduced.  Whether this was a successful way of preserving the value of the glossary is an important question I'd like reviewers to consider.

To give a rough idea of how the new index compares to the glossary, the makeindex output reports that there are 508 entries in the index, vs 71 uses of `\glossaryitem` in the glossary.  The big difference is what I believe that the index will have a much higher probability of successfully helping readers find what they need, and hence in the end become much more used than the current glossary.

I also found some outdated or conflicting statements in the glossary, and as long as the glossary isn't used much, it is hard to detect such things.  An index doesn't have this problem.  Once the index becomes a natural part of reading the document, I think it will also come natural to keep it in a well maintained state.

Finally, I must admit that I'm not sure I'm familiar with the kind of latex-glossary with back-links you are referring to.  Instead of trying to guess this now, I'll try to solve the CI problems, so that you can see what is currently done.